ios/keyman/Keyman/Keyman/KMWebBrowser/WebBrowserViewController.swift

```diff
@@ -0,0 +1,281 @@
+
+  func loadAddress(_ sender: Any, event: UIEvent) {
+    guard let urlString = addressField?.text, var url = URL(string: urlString) else {
+      NSLog("Attempting to load invalid URL: %@", addressField?.text ?? "nil")
+      return
+    }
+    if url.scheme == nil {
+      url = URL(string: "http://\(urlString)") ?? url
+    }
+    let request = URLRequest(url: url)
+    webView.loadRequest(request)
+  }
+
+  func updateAddress(_ request: URLRequest) {
+    let url: URL? = request.mainDocumentURL
+    addressField.text = url?.absoluteString
+    let userData = UserDefaults.standard
+    userData.set(url?.absoluteString, forKey: "KMWebBrowserLastURL")
+    userData.synchronize()
+  }
+
+  func refresh(_ sender: Any) {
+    webView.reload()
+  }
+
+  func stop(_ sender: Any?) {
+    webView.stopLoading()
+  }
+
+  @IBAction func back(_ sender: Any) {
+    webView.goBack()
+  }
+
+  @IBAction func forward(_ sender: Any) {
+    webView.goForward()
+  }
+
+  @IBAction func bookmarks(_ sender: Any) {
+    let bookmarksVC = BookmarksViewController()
+    bookmarksVC.webBrowser = self
+    present(bookmarksVC, animated: true, completion: nil)
+  }
+
+  @IBAction func globe(_ sender: Any) {
+    KMManager.sharedInstance().showKeyboardPicker(in: self, shouldAddKeyboard: false)
+  }
+
+  func close(_ sender: Any?) {
+    dismiss(animated: true, completion: nil)
+  }
+
+  // MARK: - UIWebViewDelegate methods
+  func webView(_ webView: UIWebView,
+               shouldStartLoadWith request: URLRequest,
+               navigationType: UIWebViewNavigationType) -> Bool {
+    updateAddress(request)
+    return true
+  }
+
+  func webViewDidStartLoad(_ webView: UIWebView) {
+    UIApplication.shared.isNetworkActivityIndicatorVisible = true
+    updateButtons()
+  }
+
+  func webViewDidFinishLoad(_ webView: UIWebView) {
+    UIApplication.shared.isNetworkActivityIndicatorVisible = false
+    appendCSSFontFamily()
+    updateButtons()
+  }
+
+  func webView(_ webView: UIWebView, didFailLoadWithError error: Error) {
+    UIApplication.shared.isNetworkActivityIndicatorVisible = false
+    updateButtons()
+    let alert = UIAlertView(title: "Cannot Open Page",
+                            message: error.localizedDescription,
+                            delegate: self,
+                            cancelButtonTitle: "OK",
+                            otherButtonTitles: "")
+    alert.show()
+  }
+
+  func updateButtons() {
+    refreshButton.isHidden = webView.isLoading
+    stopButton.isHidden = !webView.isLoading
+    backButton.isEnabled = webView.canGoBack
+    forwardButton.isEnabled = webView.canGoForward
+  }
+
+  func appendCSSFontFamily() {
+    let jsStr: String = "var style = document.createElement('style');" +
+      "style.type = 'text/css';" +
+      "style.innerHTML = '*{font-family:\"\(fontFamily)\" !important;}';" +
+    "document.getElementsByTagName('head')[0].appendChild(style);"
+    webView?.stringByEvaluatingJavaScript(from: jsStr)
+  }
+
```

Comment:
The Obj-C version had a `setFontFamily` function that seems to be missing; it would have appeared at this point in the file.

Reply:
`setFontFamily()` is generated for `var fontFamily`. The difference is that `newFontFamily` isn't updated in `setFontFamily()` as it was originally. The only thing that affects is that the webview might be reloaded an additional time in `keyboardPickerDismissed()` compared to before.

I don't think that's bad behaviour so that the webview can pick up the new font family. Or if anything, the webview should be reloaded after the font family is set.